cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -419,5 +419,53 @@ CUDA_HOST_DEVICE_CALLABLE constexpr decltype(auto) type_dispatcher(cudf::data_ty
   }
 }
 
+namespace {
+template <typename T1>
+struct dispatch_second_type_impl {
+#pragma nv_exec_check_disable
+  template <typename T2, typename F, typename... Ts>
+  CUDA_HOST_DEVICE_CALLABLE decltype(auto) operator()(F&& f, Ts&&... args) const noexcept
+  {
+    return f.template operator()<T1, T2>(std::forward<Ts>(args)...);
+  }
+};
+
+template <template <cudf::type_id> typename IdTypeMap>
+struct dispatch_second_type {
+#pragma nv_exec_check_disable
+  template <typename T1, typename F, typename... Ts>
+  CUDA_HOST_DEVICE_CALLABLE decltype(auto) operator()(cudf::data_type type2,
+                                                      F&& f,
+                                                      Ts&&... args) const noexcept
```

Comment:
If this function is going to be `noexcept`, it has to be conditional upon if the callable is `noexcept`.

Reply:
That was copied from the aggregation dispatcher -- it probably makes sense to remove the `noexcept` here.